### ISSUE TITLE
SystmeJS will not load deps of angular

### ISSUE THREAD

**Issue by gilamran** [systemjs/systemjs#518]
For the following config, SystemJs will NOT load all the needed files (jquery).

```javascript
System.defaultJSExtensions = true;
System.config({
	meta: {
		'jquery': {
			format: 'global',
			exports: 'jquery'
		},
		'angular': {
			format: 'global',
			exports: 'angular',
			deps: ['jquery']
		},
		'Application': {
			deps: ['angular']
		}
	}
});
System.import('Application');
```

This issue is specific to Angular...

**Comment by guybedford** [comment]
`deps` is only supported for the global module format. If using ES6, load angular as a normal dependency with module syntax:

```javascript
import angular from 'angular';
// ...
```

**Comment by gilamran** [comment]
I'm using TypeScript and TypeScript will returns a compilation error on this import, because in my case "angular" is a global variable... 

In TypeScript, I can't do "import angular from 'angular'" because there's no "angular" module. The angular.d.ts (The definitions file) defines a global variable named "angular".

What am I suppose to do in this situation?

**Comment by milkshakeuk** [comment]
you could try a small alteration to the angulat.d.ts so that tsc understands the es6 module export:

change this:
```javascript
declare module 'angular' {
    export = angular;
}
```
to this: 
```javascript
declare module 'angular' {
    export default angular;
}
```

or you could use typescripts original syntax for importing modules:

```javascript
var angular = require('angular');
```

**Comment by gilamran** [comment]
I've changed the configuration to look like this (Removed the angular from the deps):
```javascript
System.defaultJSExtensions = true;
System.config({
	meta: {
		'jquery': {
			format: 'global',
			exports: 'jquery'
		},
		'angular': {
			format: 'global',
			exports: 'angular',
			deps: ['jquery']
		},
		'Application': {
			deps: []
		}
	}
});
System.import('Application');
```

And the Application.js to this (Using require to get angular):
```javascript
var angular = require('angular');
angular.module("MyAppModule", []);
```

I'm not getting **Uncaught TypeError: angular.module is not a function**

**Comment by milkshakeuk** [comment]
sorry my bad with a typo the syntax is:

```javascript
import angular = require('angular');
```

**Comment by gilamran** [comment]
This syntax is not good because angular.js is not a typescript module...
Even if I use a regular javascript file (Application.js), this doesn't work...

**Comment by milkshakeuk** [comment]
This is not the case, prior the typescript 1.5 external modules such as angularjs, underscore,  jquery could be imported using that syntax because they support AMD, and that syntax the is still supported alongside the new es6 style imports.

Of course it all depends on your typescript compiler options like what ES Target and which module type in your compile flags.

**Comment by gilamran** [comment]
Let's leave TypeScript aside for now.
This syntax is not working in javascript either.

But the real question is: why can't I define angular as a global module and make my app dependent on it?
When I mark angular in the deps of my application, it will get loaded, but its own dependencies will not...

**Comment by guybedford** [comment]
Dependencies are declared via module formats. Each module file belongs to one of the module formats - ES6, System.register, AMD, CommonJS or global. Each module format has its own syntax for declaring dependencies. It is only through this means that you can create the dependency graph. If a dependency is not being loaded, the check is simply to look at the module format of the parent, and ensure the parent is declaring its dependency properly on the child.

**Comment by gilamran** [comment]
ok, so according to what you are saying, I'm using only global packages. and the configuration looks like this:

```javascript
System.defaultJSExtensions = true;
System.config({
    meta: {
        'jquery': {
            format: 'global',
            exports: 'jquery'
        },
        'angular': {
            format: 'global',
            exports: 'angular',
            deps: ['jquery']
        },
        'Application': {
            format: 'global',
            deps: ['angular']
        }
    }
});
System.import('Application');
```

As you can see the Application requires 'angular' and angular requires 'jquery'. but with this configuration angular will be loaded, but jquery will **NOT** get loaded...
What am I doing wrong?

**Comment by guybedford** [comment]
You need to specify angular in the meta by its exact name, not its map name - that is `path/to/angular.js`.

**Comment by gilamran** [comment]
Yeap, that was the problem, and the current configuration looks like this:

```javascript
System.defaultJSExtensions = true;
System.config({
    meta: {
        'bower_components/jquery/dist/jquery': {
            format: 'global',
            exports: 'jquery'
        },
        'bower_components/angular/angular': {
            format: 'global',
            exports: 'angular',
            deps: ['bower_components/jquery/dist/jquery']
        },
        'Application': {
            format: 'global',
            deps: ['bower_components/angular/angular']
        }
    }
});
System.import('Application');
```

Is there a better way to define global modules with the full path? the "map" feature is good, but can't be used on the meta...

**Comment by guybedford** [comment]
Actually it is possible to use the map name in meta as of SystemJS 0.17, but the name still needs to match at the normalization level.

**Comment by gilamran** [comment]
Well, this will not work:

```javascript
System.defaultJSExtensions = true;
System.config({
    meta: {
        'jquery': {
            format: 'global',
            exports: 'jquery'
        },
        'angular': {
            format: 'global',
            exports: 'angular',
            deps: ['jquery']
        },
        'Application': {
            deps: ['angular']
        }
    },
	path: {
		'jquery': 'bower_components/jquery/dist/jquery.js',
		'angular': 'bower_components/angular/angular.js'
	}
});
System.import('Application');
```

**Comment by guybedford** [comment]
If you put the map before the meta it will work.

**Comment by gilamran** [comment]
OMG! The order of things **do** matter...

**Comment by guybedford** [comment]
Thanks, I've added a change in the latest release to make configuration order no longer matter.

**Comment by jclif** [comment]
Is this fix in 0.16?
